### DISCUSSION TITLE
Herança na SqlConnection

### DISCUSSION THREAD

**Discussion by pedro-mileipp** [balta-io/forum#249] [Tira Dúvidas] [answered]
- Boa tarde, prezado! Cursod de Dapper, Repositórios especializados
- Por que no trecho abaixo há necessidade de chamar o método pai, já que a no Main será passado o parâmetro da mesma forma e isso não resultaria no mesmo funcionamento usando ou não o base?
```csharp
namespace Blog.Repositories
{
    public class UserRepository : Repository<User>
    {
        private readonly SqlConnection _connection;
        public UserRepository(SqlConnection connection) : base(connection)
            => _connection = connection;
    }
}
```

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @pedro-mileipp, como vai?

Conforme mostro no curso de orientação a objetos, quando passamos um objeto para classe pai, usamos o `base`.
Neste caso como precisamos manter o objeto também na classe filho, além de passar no base, deixamos no `_connection`.

Uma solução (Talvez melhor) seria deixar o objeto `SqlConnection` do repositório base como `public` ou `protected` para que possa ser acessado pelos filhos.
